pyproject.toml

```diff
@@ -61,15 +61,14 @@ dev = [
 docs = [
     "canonical-sphinx[full]~=0.2",
     "pyspelling",
-    "autodoc-pydantic~=2.0",
     "sphinx-autobuild~=2024.2",
-    "sphinx-pydantic~=0.1",
     "sphinx-sitemap>=2.6.0",
     "sphinx-toolbox~=3.5",
     "sphinx-lint~=1.0",
     "sphinxcontrib-details-directive",
     "sphinxext-rediraffe==0.2.7",
     "matplotlib",
+    "pydantic-kitbash>=0.0.6",
```

Comment:
I think our working policy is actually to pin Kitbash. https://github.com/canonical/snapcraft/blob/b511004afbba6282bc9371aa7f0584374a4ca900/pyproject.toml#L98

CC @jahn-junior